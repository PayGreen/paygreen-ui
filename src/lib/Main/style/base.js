import { css } from 'styled-components';

const isOpenStyle = css`
    @media ${props => props.theme.screen.max.md} {
        margin-left: 100%;
        box-shadow: 0 0 0 transparent;
    }

    @media ${props => props.theme.screen.min.md} {
        margin-left: ${props => props.theme.grid.sidebar};
        width: calc(100% - ${props => props.theme.grid.sidebar});
    }
`;

export { isOpenStyle };
