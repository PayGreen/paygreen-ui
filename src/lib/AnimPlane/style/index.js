import styled from 'styled-components';
import {
    planeAnimationDuration,
    dotsDelay,
    dotsTransitions,
    activeStyle,
} from './base';

const AnimPlaneBase = styled.div`
    position: relative;
    display: flex;
    justify-content: center;
    align-items: center;

    .icon {
        position: absolute;
        transform-origin: center;
        transform: scale(0);
        transition: all ${props => props.theme.transition.sm};
    }

    .illu {
        width: 100%;
        max-width: ${props => props.theme.imageSize.sm};
        fill: ${props => props.theme.color[props.colorTheme].main};

        .plane {
            transition: opacity ${props => props.theme.transition.sm};
            transition-delay: ${props => props.dotsCount * dotsDelay}s;
            transform-origin: center;
            animation-fill-mode: forwards;
            animation-delay: ${props => props.dotsCount * dotsDelay}s;
            animation-duration: ${planeAnimationDuration}s;
            animation-timing-function: linear;

            .normal {
                opacity: 0.4;
            }

            .light {
                opacity: 0.2;
            }
        }

        .dots {
            opacity: 0.4;

            path {
                opacity: ${props => (props.isActive ? 0 : 1)};
                transition: all 0;
                ${props => dotsTransitions(props.dotsCount)};
            }
        }
    }

    ${props => (props.isActive ? activeStyle : null)};
`;

export { AnimPlaneBase };
