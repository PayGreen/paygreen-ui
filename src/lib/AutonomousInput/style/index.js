import styled from 'styled-components';

const AutonomousInputBase = styled.div`

`;

export { AutonomousInputBase };