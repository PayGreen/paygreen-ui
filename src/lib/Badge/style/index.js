import styled from 'styled-components';
import { math } from 'polished';
import { badgeSize, originalStyle, reverseStyle } from './base';

const BadgeBase = styled.span`
    display: flex;
    justify-content: center;
    align-items: center;
    width: fit-content;
    border-radius: ${props => props.theme.radius.sm};
    height: ${props => math(props.theme.font.size[props.textSize] + ' + 1rem')};
    font-size: ${props => props.theme.font.size[props.textSize]};
    text-align: center;
    text-transform: uppercase;
    letter-spacing: ${props => props.theme.font.spacing};
    font-weight: ${props => props.theme.font.weight.bold};

    ${props => badgeSize[props.textSize]};
    ${props => (props.colorType === 'original' ? originalStyle : reverseStyle)};
`;

export { BadgeBase };
