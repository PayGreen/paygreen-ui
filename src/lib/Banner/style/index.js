import styled from 'styled-components';
import { waveOptions } from '../../../shared/constants';
import { color, waveScale } from './constants';

const BannerBase = styled.div`
    position: relative;
    margin-top: ${props => props.topStyle !== waveOptions.none ? props.theme.wave.double : 0};
    margin-bottom: ${props => props.bottomStyle !== waveOptions.none ? props.theme.wave.double : 0};
    padding: .1px 0;
    background-image: linear-gradient(
        ${props => color.first[props.gradient]},
        ${props => color.second[props.gradient]}
    );

    .waves {
        position: absolute;
        left: 0;

        .light {
            opacity: .5;
        }

        &:first-child {
            fill: ${props => color.first[props.gradient]};
            bottom: 100%;
            transform: scale(${props => waveScale[props.topStyle]}, ${waveScale.top});
            margin-bottom: -1px;
        }
        
        &:last-child {
            fill: ${props => color.second[props.gradient]};
            top: 100%;
            transform: scale(${props => waveScale[props.bottomStyle]}, ${waveScale.bottom});
            margin-top: -1px;
        }
    }
`;

export { BannerBase };