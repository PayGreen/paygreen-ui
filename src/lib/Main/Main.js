import React from 'react';
import { MainBase } from './style';

const Main = props => {
    return <MainBase {...props}>{props.children}</MainBase>;
};

export default Main;