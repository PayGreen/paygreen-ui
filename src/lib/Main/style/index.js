import styled from 'styled-components';
import { transparentize } from 'polished';
import { TopbarBase } from '../../Topbar/style';
import { isOpenStyle } from './base';

const Box = styled.div``;

const MainBase = styled.div`
    position: relative;
    height: 100vh;
    width: 100%;
    display: grid;
    grid-template-rows: auto 1fr;
    grid-template-areas: 'topbar' 'box';
    overflow-y: hidden;
    background-color: ${props => props.theme.wab.white00};
    box-shadow: ${props =>
        props.theme.shadow.size.lg +
        ' ' +
        transparentize(props.theme.shadow.opacity.lg, props.theme.wab.black00)};
    transition: all ${props => props.theme.transition.sm};

    & > ${TopbarBase} {
        grid-area: topbar;
    }

    & > ${Box} {
        grid-area: box;
        overflow-y: auto;
    }

    ${props => (props.isOpen ? isOpenStyle : null)};
`;

export { Box, MainBase };