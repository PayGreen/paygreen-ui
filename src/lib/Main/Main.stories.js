import React from 'react';
import { storiesOf } from '@storybook/react';
import { withKnobs, boolean } from '@storybook/addon-knobs';
import { folder, spaceOptions } from '../../shared/constants';
import Main from './Main';
import Topbar from '../Topbar/Topbar'
import MenuHamburger from '../MenuHamburger/MenuHamburger';
import Logo from '../Logo/Logo';
import { logo } from '../Logo/sample/logo';

const mainIsOpenLabel = 'Open main';
const mainMenuIsOpenLabel = 'Open primary menu';

storiesOf(folder.nav + 'Main', module)
    .addDecorator(withKnobs)
    .add('Main', () => (
        <Main isOpen={boolean(mainIsOpenLabel, true)}>
                <Topbar>
            <div className="main-nav">
                <MenuHamburger isOpen={boolean(mainMenuIsOpenLabel, true)} />
                <a href="#">
                    <Logo
                        hasBaseline={false}
                        hasHoverRight={true}
                        blockWidth={spaceOptions.sm}
                    >
                        {logo}
                    </Logo>
                </a>
            </div>
        </Topbar>
        </Main>
    ));
