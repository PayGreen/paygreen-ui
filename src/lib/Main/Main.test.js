import React from 'react';
import TestRenderer from 'react-test-renderer';
import { ThemeDefault } from '../../theme';
import Main from './Main';
import MenuHamburger from '../MenuHamburger/MenuHamburger';
import Topbar from '../Topbar/Topbar';
import { Box } from './style';

it('renders without crashing', () => {
    const main = TestRenderer.create(
        <Main theme={ThemeDefault}>
            <Topbar theme={ThemeDefault}>
                <MenuHamburger theme={ThemeDefault} />
            </Topbar>
            <Box />
        </Main>,
    );
    expect(main.toJSON()).toMatchSnapshot();
});
