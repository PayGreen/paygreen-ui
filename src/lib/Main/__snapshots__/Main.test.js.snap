// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<div
  className="sc-htpNat iHbVbz"
>
  <header
    className="sc-bdVaJa icKNKz"
  >
    <button
      className="sc-bxivhb WUaOj"
    >
      <span />
      <span />
      <span />
    </button>
  </header>
  <div
    className="sc-bwzfXH hIddlc"
  />
</div>
`;
